src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -483,6 +483,10 @@ for (var type in topLevelEventsToDispatchConfig) {
 var ON_CLICK_KEY = keyOf({onClick: null});
 var onClickListeners = {};
 
+function getDictionaryKey(inst) {
+  return '.' + inst._rootNodeID;
+}
+
```

Comment:
@gaearon I have a couple of questions about this solution, could you please explain this.

- I'm not sure about how this changes don't break the current behaviour. In the previous code the key was just `inst._rootNodeId` and not it's `'.' + inst._rootNodeId`. How this could be the same thing as there're no other changes?
- Why does it improve performance?

Reply:
I agree

Reply:
Why not throw a null pointer exception to the engine itself? That way you could get 50 caliber speeds.